src/RoslynPad/DocumentViewModel.cs

```diff
@@ -19,16 +19,76 @@ internal class DocumentViewModel : NotificationObject
         private DocumentViewModel(MainViewModel mainViewModel)
         {
             MainViewModel = mainViewModel;
-            var defaultPath = GetDefaultPath();
-            Directory.CreateDirectory(defaultPath);
-            Path = defaultPath;
+            Path = GetUserDocumentPath();
+            Directory.CreateDirectory(Path);
             IsFolder = true;
         }
 
-        public string Path { get; }
+        public string Path { get; internal set; }
+
         public MainViewModel MainViewModel { get; }
         public bool IsFolder { get; }
 
+        internal static string GetUserDocumentPath()
+        {
+            var userDefinedPath = Properties.Settings.Default.DocumentPath;
+            return !string.IsNullOrEmpty(userDefinedPath) && System.IO.Directory.Exists(userDefinedPath)
+                ? userDefinedPath
+                : GetDefaultDocumentPath();
+        }
+
+        private static string GetDefaultDocumentPath()
+        {
+            return System.IO.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "RoslynPad");
+        }
+
+        public async void EditUserDocumentPath()
+        {
+            // TODO: Review expected behavior. 
+            // Currently rebinding documents pane.
+            // Dialog requesting restart? Auto-restart app? Rebind documents pane? What happens to open documents?
+
+            var dialog = new System.Windows.Forms.FolderBrowserDialog()
+            {
+                ShowNewFolderButton = false,
+                SelectedPath = GetUserDocumentPath()
+            };
+
+            var result = dialog.ShowDialog();
+            if (result == System.Windows.Forms.DialogResult.OK)
+            {
+                await MainViewModel.CloseAllDocuments();
```

Comment:
We definitely don't want to close documents just because we changed folders. The current documents will continue to be saved in the original location, and that's fine.